<launch>
    <node name="hsv_filter_node" pkg="vision" type="hsv_filter_node" output="screen">
            <param name="h_low" value="0" type="int" />
            <param name="s_low" value="0" type="int" />
            <param name="v_low" value="0" type="int" />
            <param name="h_high" value="248" type="int" />
            <param name="s_high" value="248" type="int" />
            <param name="v_high" value= "67" type="int" />
            <remap from="/camera/image_raw" to="/uwsim/camera2"/>
    </node>

    <node name="gate_detect" pkg="gate_detect" type="gate_detect_node" output="screen">
            <param name="cannyLow" value="50" type="int" />
            <param name="houghLinesThreshold" value="100" type="int" />
            <param name="houghLinesMinLength" value="125" type="int" />
            <param name="houghLinesMaxLineGap" value="150" type="int" />
            <param name="poleMax" value="50" type="int" />
            <param name="interpolationConstant1" value="81.88" type="double" />
            <param name="interpolationConstant2" value="-.92791" type="double" />
    </node>

    <node name="hsv_filter_node1" pkg="vision" type="hsv_filter_node" output="screen">
        <param name="h_low" value="163" type="int" />
            <param name="s_low" value="0" type="int" />
            <param name="v_low" value="0" type="int" />
            <param name="h_high" value="262" type="int" />
            <param name="s_high" value="255" type="int" />
            <param name="v_high" value= "160" type="int" />
            <remap from="/camera/image_raw" to="/uwsim/camera1"/>
            <remap from="/vision/output" to="/vision/output1"/>
    </node>

    <node name="line_detect" pkg="line_detect" type="line_detect_node" output="screen">
            <remap from="camera_output" to="/vision/output1"/>
    </node>

    <node name="world_state_node" pkg="worldstate" type="world_state_node" output="screen">
    </node>

    <node name="decision_node" pkg="decision" type="decision_node" output="screen">
        <remap from="worldstate" to="/worldstate/output"/>
        <remap from="gate_location" to="/gateDetect/output"/>
        <remap from="/decision_node/output" to="/g500/twist"/>
    </node>
    <rosparam command="load" file="$(find decision)/constants/constants.yaml" ns="global_constants" />
</launch>
